include/RAJA/internal/Iterators.hpp

```diff
@@ -67,32 +67,18 @@ namespace RAJA
 namespace Iterators
 {
 
-
-// Helpers
-
-
-template <typename Container>
-using IteratorCategoryOf =
-    typename std::iterator_traits<typename std::remove_reference<Container>::
-                                      type::iterator>::iterator_category;
-
-template <typename Container>
-using OffersRAI = std::is_base_of<std::random_access_iterator_tag,
-                                  IteratorCategoryOf<Container>>;
-
 // Containers
 
 template <typename Type,
           typename DifferenceType = std::ptrdiff_t,
           typename PointerType = Type*>
-class base_iterator : public std::iterator<std::random_access_iterator_tag,
-                                           Type,
-                                           DifferenceType>
-{
-public:
-  using difference_type =
-      typename std::iterator<std::random_access_iterator_tag,
-                             Type>::difference_type;
+struct base_iterator {
+
+  using value_type = Type;
+  using difference_type = DifferenceType;
+  using pointer = value_type*;
+  using reference = value_type&;
+  using iterator_category = std::random_access_iterator_tag;
```

Comment:
Newer C++ is making `iterator_traits` go the way of the dodo -- prefer to manually list all type aliases